src/engine/routing_algorithms/alternative_path_mld.cpp

```diff
@@ -429,13 +429,83 @@ filterAnnotatedRoutesByStretch(RandIt first, RandIt last, const InternalRouteRes
 
     BOOST_ASSERT(shortest_route.is_valid());
 
-    const auto shortest_route_duration = shortest_route.duration();
-    const auto scaled_at_most_longer_by =
-        scaledAtMostLongerByFactorBasedOnDuration(shortest_route_duration);
-    const auto stretch_duration_limit = (1. + scaled_at_most_longer_by) * shortest_route_duration;
+    static constexpr double kStretchScale = 1.2f;
+
+    //    const auto shortest_route_duration = shortest_route.duration();
+    //    const auto scaled_at_most_longer_by =
+    //        scaledAtMostLongerByFactorBasedOnDuration(shortest_route_duration);
+    //    const auto stretch_duration_limit = (1. + scaled_at_most_longer_by) *
+    //    shortest_route_duration;
+
+    const auto over_duration_limit = [&](const InternalRouteResult &route) {
```

Comment:
Could you add a high-level description comment of what the following check does and why we need it